code/game/objects/machinery/nuclearbomb.dm

```diff
@@ -6,334 +12,244 @@ GLOBAL_LIST_EMPTY(nukes_set_list)
-		if (href_list["auth"])
-			if (src.auth)
-				src.auth.loc = src.loc
-				src.yes_code = 0
-				src.auth = null
-			else
-				var/obj/item/I = usr.get_active_held_item()
-				if (istype(I, /obj/item/disk/nuclear))
-					if(usr.drop_held_item())
-						I.forceMove(src)
-						auth = I
-		if (src.auth)
-			if (href_list["type"])
-				if (href_list["type"] == "E")
-					if (src.code == src.r_code)
-						src.yes_code = 1
-						src.code = null
-					else
-						src.code = "ERROR"
-				else
-					if (href_list["type"] == "R")
-						src.yes_code = 0
-						src.code = null
-					else
-						src.code += text("[]", href_list["type"])
-						if (length(src.code) > 5)
-							src.code = "ERROR"
-			if (src.yes_code)
-				if (href_list["time"])
-					var/time = text2num(href_list["time"])
-					src.timeleft += time
-					src.timeleft = min(max(round(src.timeleft), 60), 600)
-				if (href_list["timer"])
-					if (src.timing == -1.0)
-						return
-					if (src.safety)
-						to_chat(usr, "<span class='warning'>The safety is still on.</span>")
-						return
-					src.timing = !( src.timing )
-					if (src.timing)
-						if(!src.lighthack)
-							src.icon_state = "nuclearbomb2"
-						if(!src.safety)
-							GLOB.nukes_set_list |= src
-						else
-							GLOB.nukes_set_list -= src
-					else
-						GLOB.nukes_set_list -= src
-						if(!src.lighthack)
-							src.icon_state = "nuclearbomb1"
-				if (href_list["safety"])
-					src.safety = !( src.safety )
-					if(safety)
-						src.timing = 0
-						stop_processing()
-						GLOB.nukes_set_list -= src
-				if (href_list["anchor"])
-
-					if(removal_stage == 5)
-						src.anchored = FALSE
-						visible_message("<span class='warning'> \The [src] makes a highly unpleasant crunching noise. It looks like the anchoring bolts have been cut.</span>")
-						return
-
-					src.anchored = !( src.anchored )
-					if(src.anchored)
-						visible_message("<span class='warning'> With a steely snap, bolts slide out of [src] and anchor it to the flooring.</span>")
-					else
-						visible_message("<span class='warning'> The anchoring bolts slide back into the depths of [src].</span>")
 
-		for(var/mob/M in viewers(1, src))
-			if ((M.client && M.interactee == src))
-				src.attack_hand(M)
-	else
-		usr << browse(null, "window=nuclearbomb")
-		return
-	return
+	usr.set_interaction(src)
+	
+	if(href_list["r_auth"])
+		if (r_auth)
+			r_auth.forceMove(loc)
+			has_auth = FALSE
+			r_auth = null
+		else
+			var/obj/item/I = usr.get_active_held_item()
+			if (!istype(I, /obj/item/disk/nuclear/crash/red))
+				return
+			if(!usr.drop_held_item())
+				return
+			I.forceMove(src)
+			r_auth = I
+	if(href_list["g_auth"])
+		if (g_auth)
+			g_auth.forceMove(loc)
+			has_auth = FALSE
+			g_auth = null
+		else
+			var/obj/item/I = usr.get_active_held_item()
+			if(!istype(I, /obj/item/disk/nuclear/crash/green))
+				return
+			if(!usr.drop_held_item())
+				return
+			I.forceMove(src)
+			g_auth = I
+	if(href_list["b_auth"])
```

Comment:
Could cominbe it into an assoc list of auth disks and then make each one a different paramter rather than  having  basically the same code three times